### PR TITLE
[Maistra-566] Make prometheus image configurable

### PR DESCRIPTION
**Pull request by brian-avery**

### CODE REVIEW

**Comment by brian-avery** [comment]
https://issues.jboss.org/browse/MAISTRA-566

File: tmp/build/patch-charts.sh [hunk 1]
```diff
@@ -157,6 +157,9 @@ function patchTemplates() {
           -e 's/tag:.*$/tag: '${MAISTRA_VERSION}'/' \
           -e 's/image: *proxy_init/image: proxy-init-ubi8/' \
           -e 's/image: *proxyv2/image: proxyv2-ubi8/' ${HELM_DIR}/istio/values.yaml ${HELM_DIR}/istio-init/values.yaml
+
+		# TODO: currently we're appending. When this comes down from upstream, we'll want to replace this with a substitution
+		sed -i -e "\$aimage: prometheus-ubi8" ${HELM_DIR}/istio/charts/prometheus/values.yaml
```
Comment (dmitri-d) [inline tmp/build/patch-charts.sh:162]: perhaps append this at the top of values.yaml, after "tag" perhaps? Just so it's more visible?

**Comment by dmitri-d** [comment]
This looks good to me, the only question I have is whether we should keep all prometheus-related patches in patch-prometheus.sh?

**Comment by luksa** [comment]
LGTM. 

I need these changes right now, so I'll merge this and address @dmitri-d's comments in a followup PR

File: tmp/build/patch-prometheus.sh [hunk 2]
```diff
@@ -49,16 +49,8 @@ function prometheus_patch_deployment() {
...
-  else
-    sed -i -r -e 's/image:(.*)prometheus:/image:\1prometheus-rhel8:/' ${HELM_DIR}/istio/charts/prometheus/templates/deployment.yaml
-  fi
-
-  sed -i -e '/image.*prometheus/,/args:/ {
-      /args:/ a\
-            - --storage.tsdb.path=/prometheus
-    }' ${HELM_DIR}/istio/charts/prometheus/templates/deployment.yaml
+  sed -i -r -e 's/image:(.*)prometheus:/image:\1{{ \.Values\.image }}:/' ${HELM_DIR}/istio/charts/prometheus/templates/deployment.yaml
+	sed "/storage.tsdb.retention.*/a\ \ \ \ \ \ \ \ \ \ \ \ - \'--storage.tsdb.path=/prometheus\'" deployment.yaml
```
Comment (luksa) [inline tmp/build/patch-prometheus.sh:53]: Path to deployment.yaml is incorrect here. I've fixed this in https://github.com/Maistra/istio-operator/pull/116

**Comment by luksa** [comment]
Follow-up PR: https://github.com/Maistra/istio-operator/pull/117

**Comment by jwendell** [comment]
Jira issue number is wrong

**Pull request merged**